src/compiler/checker.ts

```diff
@@ -14881,16 +14920,24 @@ namespace ts {
 
             // Initialize global symbol table
             forEach(host.getSourceFiles(), file => {
-                if (!isExternalModule(file)) {
+                if (!isExternalOrCommonJsModule(file)) {
                     mergeSymbolTable(globals, file.locals);
                 }
             });
 
             // Initialize special symbols
+            if (compilerOptions.allowNonTsExtensions) {

```

Comment:
Seems odd to have this determine whether `require` is a special global. I think it should either (a) always be a special global or (b) only be a special global in JavaScript files.